crates/nu-protocol/src/config/output.rs

```diff
@@ -24,3 +26,116 @@ impl UpdateFromValue for ErrorStyle {
         config_update_string_enum(self, value, path, errors)
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, IntoValue, Serialize, Deserialize)]
+pub enum UseAnsiColoring {
+    #[default]
+    Auto,
+    True,
+    False,
+}
+
+impl UseAnsiColoring {
+    /// Determines whether ANSI colors should be used.
```

Comment:
Great documentation here. Thanks for taking the time to spell out exactly what it does.